### PR TITLE
Use model.matrix() in check_outliers instead of get_predictors()

### PR DESCRIPTION
**Pull request by bwiernik**
ncol(data) needs to refer to the number for terms in the model function. The current approach is incorrect because it (1) doesn't separate factor variables into levels and (2) doesn't include the intercept.

Closes https://github.com/easystats/performance/issues/268

### CODE REVIEW

File: R/check_outliers.R [hunk 1]
```diff
@@ -184,8 +184,7 @@ check_outliers.default <- function(x, method = c("cook", "pareto"), threshold =
   method <- match.arg(method, c("zscore", "iqr", "cook", "pareto", "mahalanobis", "robust", "mcd", "ics", "optics", "iforest", "lof"), several.ok = TRUE)
 
   # Remove non-numerics
-  data <- insight::get_predictors(x)
-  data <- data[, sapply(data, is.numeric), drop = FALSE]
+  data <- model.matrix(x)
```
Comment (strengejacke) [inline R/check_outliers.R]: please use `stats::model.matrix()` (and add `@importFrom stats model.matrix`).

The only problem I see is that `model.matrix` is less stable than `get_predictors()`. E.g., `nlme::lme()` has no model.matrix method:

``` r
library(nlme)
fm1 <- lme(distance ~ age, data = Orthodont) # random is ~ age
model.matrix(fm1)
#> Error in eval(predvars, data, env): object 'distance' not found
```

<sup>Created on 2021-04-15 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

@DominiqueMakowski Maybe we should also think about implementing a more robust `model.matrix()` in _insight_?

**Review by strengejacke — CHANGES_REQUESTED** [review]

File: R/check_outliers.R (hunk 1 above)
Comment (bwiernik) [reply R/check_outliers.R]: That last point would be a good approach.

Comment (DominiqueMakowski) [reply R/check_outliers.R]: Yes, there are multiple reasons for that (e.g., [1](https://github.com/easystats/insight/issues/316#issuecomment-793537395) and [2](https://github.com/easystats/performance/issues/186)). We can call it `get_modelmatrix` or `model_matrix`?

Comment (bwiernik) [reply R/check_outliers.R]: `get_modelmatrix` `get_model_matrix` would be more consistent.

Comment (bwiernik) [reply R/check_outliers.R]: > @importFrom stats model.matrix)

If a function is called with its namespace `stats::model.matrix`, it isn't necessary to also import it. Those are redundant.

Comment (strengejacke) [reply R/check_outliers.R]: > If a function is called with its namespace stats::model.matrix, it isn't necessary to also import it. Those are redundant.

Why? Else, those won't be updated/listed in the NAMESPACE file.

Comment (bwiernik) [reply R/check_outliers.R]: They don't need to be in the NAMESPACE file if you namespace them when you call them. That is one of the major advantages of namespacing when you call them. Also helps avoid namespace collisions.

Comment (strengejacke) [reply R/check_outliers.R]: Ok, so you don't need to tell R that there is a selective import? Or is this only true for the "core" packages (stats, utils, ...)?

Comment (strengejacke) [reply R/check_outliers.R]: I just read this chapter: https://r-pkgs.org/namespace.html
and as far as I understood, it indeed seems like you said. In this case, I would vote for using namespace in function calls, and remove the `@importFrom`.

Comment (bwiernik) [reply R/check_outliers.R]: Yeah, you only need to importFrom if you want to use the function without the namespace (eg, ggplot2 functions or some stats functions might be worth importing)

**Pull request merged**